terraform/modules/swipe-sfn-batch-queue/main.tf

```diff
@@ -68,8 +68,13 @@ resource "aws_launch_template" "swipe_batch_main" {
   # not recognize this change. We bind the launch template name to user data contents here, so any changes to user data
   # will cause the whole launch template to be replaced, forcing the compute environment to pick up the changes.
   name      = "${var.app_name}-batch-main-${local.launch_template_user_data_hash}"
-  user_data = base64encode(local.launch_template_user_data)
   tags      = var.tags
+  user_data = base64encode(concat(var.user-data-parts,
```

Comment:
I am a little confused by this, this looks like it is calling `base64encode` on an array of objects. I think this needs to be called on a string. When I try this in `terraform console`:

```terraform
base64encode([{ filename = "foo" }])
```

I get an error:

```
│ Invalid value for "str" parameter: string required.
```

Is this the intent here?

Reply:
I assumed that this was what `template_cloudinit_config` does in the rendered output but in hindsight that sounds like not a great assumption. I had hoped to avoid pasting another template from here https://github.com/chanzuckerberg/shared-infra/pull/5153/files#diff-03f83f2902ab11032bdcb34beefa58f27ead92a7f43288f67ee2db51e859ef65R128 but that might end up being the safest way